crates/e2e/src/setup/services.rs

```diff
@@ -51,6 +51,14 @@ impl<'a> Services<'a> {
         }
     }
 
+    pub async fn new_with_timeout(contracts: &'a Contracts, timeout: Duration) -> Services<'a> {
```

Comment:
Functions like `with_something()` usually indicate a builder pattern which also seems more appropriate in case we ever have to add more optional configurations you want to opt into.
So specifically I would recommend a signature like:
```
pub fn with_http_timeout(self, timeout: Duration) -> Self;
```
That way you can easily chain function calls modifying `Self`.

Reply:
I was going to do something like this originally but then realized that `reqwest::Client` does not allow you to change the timeout after it is built. So, I would need to build a small `ServicesBuilder` struct and impl, which I don't mind doing of course if you prefer!

FWIW, secondary constructors with something like `with_foo` are part of the official API guidelines: https://rust-lang.github.io/api-guidelines/predictability.html#constructors-are-static-inherent-methods-c-ctor

But I understand if you prefer to have a proper `ServicesBuilder` already. :)